sympy/solvers/inequalities.py

```diff
@@ -521,6 +522,8 @@ def solve_univariate_inequality(expr, gen, relational=True, domain=S.Reals, cont
                 # this might raise ValueError on its own
                 # or it might give None...
                 solns = solvify(e, gen, domain)
+                if domain is (S.Integers):
+                    solns = solvify(e, gen, S.Reals)
```

Comment:
Would it not be better to improve `solvify` so that it does the right thing when the domain is `Integers`?

Reply:
Actually, we just want all solutions so that they could be further processed integers!
As `solvify` doesn't return any solution for non-integers.
So it will be hard to take out the range for them.
E.G.
if `solvify` returns `[9,8/9] ` for reals
then for integer domain, it returns only [9]
in such a case it's hard to analyze the whole interval (solution for integers)!

Reply:
Why not just call solvify with the Reals as the domain in the first place?

Reply:
> Why not just call solvify with the Reals as the domain in the first place?

Maybe because it may hinder equations with other domains.

Reply:
This needs more careful thought. If it's okay to use a different domain when the domain is Integers then why wouldn't it be okay to do that when the domain is anything else?

At minimum `solvify` should not be called twice.

Reply:
What about , instead of calling it 2 times

```
dom = lambda d: S.Reals if d is S.Integers else domain
solns = solvify(e, gen, dom(domain))
```

Reply:
You haven't answered my question. Why is it okay to change the domain if the domain is Integers but not for any other possible domain?

This looks like a hack to me. If you can't answer the question then I don't think that this change should be made. 

I'm not going to go round and round in circles on this like in the other PRs though: if you can't answer a direct question then I am not going to review any of your PRs in future.

Reply:
> Why is it okay to change the domain if the domain is Integers but not for any other possible domain?


I've answered your question in this comment https://github.com/sympy/sympy/pull/21156#discussion_r600682918.

Actually, we just want all solutions so that they could be further processed for integers!
As `solvify` doesn't return any solution which is non-integers. ( when the domain is `Integers`)
So it will be hard to take out the range for them.

E.G.
if `solvify` returns `[9,8/9] ` for reals
then for integer domain, it returns only [9]
in such a case it's hard to analyze the whole interval (i.e. solution for integers)!

Reply:
> I'm not going to go round and round in circles on this like in the other PRs though: if you can't answer a direct question then I am not going to review any of your PRs in future.

I think you are really getting frustrated while dealing with me. For which I'm really sorry and please don't think I'm doing this intentionally, I'm just trying to do my level best : )
If you really want me to quit this platform then I think I should leave.

Reply:
I think we're both getting frustrated and it's not helping anything.

The problem I have is that I don't know enough about the `solveset` code to be able to quickly say what should be done. That means that for me to be able to review this work I need you to explain it to me in a way that I can understand. That doesn't seem to be possible so each time I ask a question we get stuck.

I'm not saying that I want you to stop contributing but I don't think it's helpful for anyone if I continue reviewing your PRs. Other people also review PRs and they are welcome to review this.

Reply:
Yep, I think that is better, someone with a better grip over `solveset` can review.
Most of the time I give a reason but I think you are not able to get it. Or maybe I'm not able to anticipate to what extent I have to explain the code and accordingly give a reason.

Btw, thanks for everything : )